### ISSUE TITLE
Failed to register: Unable to acquire bus name 'org.gnome.Extensions'

### ISSUE THREAD

**Issue by ad-on-is** [Ulauncher/Ulauncher#951]
### Environment

OS: EndeavourOS 5.15.13-arch1-1,  DE: GNOME 41.3 

### Bug description

I just updated my system and realized Ulauncher won't open the Extensions app.

- Open Ulauncher
- Type "Extensions"
- Gnome shell extensions shows up in list
- Select GNOME Extensions app
- Extensions won't open
- `journalctl` produces the following output: `Failed to register: Unable to acquire bus name 'org.gnome.Extensions'`

### Log output

```shell
`ulauncher -v` does not report any reasonable output, but `sudo journalctl -f` does
```

### Communication guidelines

- [X] I have read and agree with Ulauncher's communication guidelines.

### Not an extension issue

- [X] This issue applies to the Ulauncher application, not a specific extension.

### Latest version

- [X] I am using the latest Ulauncher release.

**Comment by friday** [comment]
Also on arch and can't reproduce this.

> Failed to register: Unable to acquire bus name 'org.gnome.Extensions'

This means it's already running, or there's some issue with dbus. It doesn't have anything to do with Ulauncher and rebooting your system will probably fix it.

> `ulauncher -v` does not report any reasonable output, but `sudo journalctl -f` does

`ulauncher -v` only works if you stop the systemd daemon (`systemctl --user stop ulauncher`) first, but journalctl should give the same info if you don't.

**Comment by ad-on-is** [comment]
> Also on arch and can't reproduce this.
> 
> > Failed to register: Unable to acquire bus name 'org.gnome.Extensions'
> 
> This means it's already running, or there's some issue with dbus. It doesn't have anything to do with Ulauncher and rebooting your system will probably fix it.
> 
> > `ulauncher -v` does not report any reasonable output, but `sudo journalctl -f` does
> 
> `ulauncher -v` only works if you stop the systemd daemon (`systemctl --user stop ulauncher`) first, but journalctl should give the same info if you don't.

No, unfortunately a restart does not solve the problem. It doesn't even start the app, if I try to open it after a reboot.

However, launching "Extensions" from within GNOME launcher, works.

**Comment by friday** [comment]
If you run `systemctl --user stop ulauncher` and then `ulauncher -v` it works?

**Comment by ad-on-is** [comment]
> If you run `systemctl --user stop ulauncher` and then `ulauncher -v` it works?

> It doesn't even start the app, if I try to open it after a reboot.

I just realized, this sentence is confusing. Ulauncher itself starts and works well, except when trying to open the "Extensions" app.

---

I now stopped Ulauncher, and reran it with `ulauncher -v`

This is the output I get, when I select "Extensions" from the drop-down list.
```
2022-01-14 10:41:35,893 | INFO | ulauncher.api.shared.action.LaunchAppAction: run() | Run application Extensions (/usr/share/applications/org.gnome.Extensions.desktop) Exec /usr/bin/gnome-extensions-app --gapplication-service
2022-01-14 10:41:35,903 | DEBUG | ulauncher.ui.windows.UlauncherWindow: show_results() | render 0 results
Failed to register: Unable to acquire bus name 'org.gnome.Extensions'
```

**Comment by friday** [comment]
I managed to reproduce it (just had to update gnome-shell, sorry).

I think it has to do with [this](https://wiki.gnome.org/HowDoI/DBusApplicationLaunching)

> If `DBusActivatable` is `true` and the desktop file name looks like a valid application ID, then the `Exec` line will be ignored and your application will be started by way of D-Bus activation instead (using the name of the desktop file as the application ID).

We use Exec directly, which should work, but this app has broken it, so I guess we have to implement the D-Bus activation method (which should be a good idea anyway).

In the meanwhile I think you can work around it like this:
`cp /usr/share/applications/org.gnome.Extensions.desktop ~/.local/share/applications/`

Then edit `~/.local/share/applications/org.gnome.Extensions.desktop` and remove the argument in `Exec` so it only says `Exec=/usr/bin/gnome-extensions-app`

**Comment by ad-on-is** [comment]
Aah, glad you were able to reproduce it.

When you switch to D-Bus activation, will I need to remove the file from `~/.local/applications` or will it still work? Otherwise, I can just leave it as is, and launch it from gnome instead, since it's not a big deal for now.

**Comment by friday** [comment]
This will be fixed soon, so you can skip that workaround if you want.

**Comment by friday** [comment]
Released as [5.14.3](https://github.com/Ulauncher/Ulauncher/releases/tag/5.14.3)

**Comment by ad-on-is** [comment]
The update (5.14.3) also just got released to AUR, and it works now.
